axelrod/moran.py

```diff
@@ -218,16 +204,15 @@ def mutate(self, index: int) -> Player:
             return self.players[index].mutate()
 
         # Assuming mutation_method == "transition"
-        if self.mutation_rate > 0:
-            # Choose another strategy at random from the initial population
-            r = self._random.random()
-            if r < self.mutation_rate:
-                s = str(self.players[index])
-                j = self._random.randrange(0, len(self.mutation_targets[s]))
-                p = self.mutation_targets[s][j]
-                return p.clone()
-        # Just clone the player
-        return self.players[index].clone()
+        if self._random.random() > self.mutation_rate:
```

Comment:
I believe but I might be remembering incorrectly that we had it this way to avoid sampling a random number if no such sample was needed. I think this helped with performance.

Reply:
Did some quick tests, and I'm seeing that drawing a random number takes 1.4 microseconds, while an if takes 2.1 microseconds.  We actually reduce an if when mutation_rate = 0, so we may be saving time.  But moreover, it doesn't seem like a meaningful optimization.

I'm not sure, though.  Should we add a performance test?

Reply:
Note that the class now uses the bulk random number generator, which should be faster at scale. This came later and may have altered the tradeoffs.

Reply:
I will add a benchmark in another change, coming soon.